src/Microsoft.DotNet.Interactive/KernelEventExtensions.cs

```diff
@@ -60,4 +62,14 @@ private static IReadOnlyCollection<Diagnostic> RemapDiagnosticsFromLanguageNode(
             )
         ).ToImmutableList();
     }
+
+    internal static void StampRoutingSlipAndLog(this KernelEvent @event, Uri uri)
```

Comment:
Why not just add the log call into the `Stamp*` methods? Does it get too verbose?

Reply:
The `Stamp*` methods don't have access to the `event` /`command`. So we need to add it to the caller of those methods.

Unless you think its ok to change the public `Stamp*` methods to pass in the `event` / `command`... But even then, we only want to log when routing slip is stamped inside the tool process - but not when it is stamped within the (de)serializer.

Reply:
The deserialization process should not call `Stamp`. If it still does that we should fix it.

Reply:
It does currently. See https://github.com/dotnet/interactive/blob/main/src/Microsoft.DotNet.Interactive/Connection/KernelCommandEnvelope.cs#L186.

To fix this, I guess `RoutingSlip` should have a constructor that accepts all deserialized uris (and tags), and we need a `KernelCommand.SetRoutingSlip` (similar to `KernelCommand.SetToken`) to initialize `KernelCommand.RoutingSlip` post deserialization. Is this what you were implying?

Reply:
Yeah, that sounds right.